tests/php/test-class-amp-dom-document.php

```diff
@@ -199,6 +199,21 @@ public function data_dom_document() {
 				'<!DOCTYPE html><html><head profile=""></head><body></body></html>',
 				'<!DOCTYPE html><html><head><meta charset="utf-8"></head><body></body></html>',
 			],
+			'amp_mustache_template_with_table'         => [
+				'utf-8',
+				'<!DOCTYPE html><html>' . $head . '<body><template type="amp-mustache"><table><tr>{{#example}}<td></td>{{/example}}</tr></table></template></body></html>',
+				'<!DOCTYPE html><html>' . $head . '<body><template type="amp-mustache"><table><tr>{{#example}}<td></td>{{/example}}</tr></table></template></body></html>',
+			],
+			'amp_mustache_template_script_with_table'  => [
+				'utf-8',
+				'<!DOCTYPE html><html>' . $head . '<body><script id="baz" type="text/plain" template="amp-mustache"><table><tr>{{#example}}<td></td>{{/example}}</tr></table></script></body></html>',
```

Comment:
Could you add a test for a `script` that had more than one child node? This will help ensure no bugs get introduced in regards to the order at restoration.

Reply:
Sure, good idea

Reply:
Applied in https://github.com/ampproject/amp-wp/pull/4333/commits/5ef39b71c5eee8c7c40d97b554385c7683db9993